Refactor script to use functions

Mudskipper875 opened the pull request:
Added two functions: One for moving/backing up the dotfiles and one for appending
the 'export variable' command to tidytilde_backups.

The functions are explained in the comments.

I'm not sure if backups for all the dotfiles are needed. Some directories like .rustup and .cargo are very big and will take a while to copy.

Mudskipper875 committed:
Refactor script to use functions

Added two functions for moving the dotfiles and appending the
'export variable' command to tidytilde_backups.

The functions are explained in the comments.

gtlsgamr commented:
The backups are necessary, at least until we have tested this script to hell and back. Looks fine otherwise.

Mudskipper875 commented:
> The backups are necessary, at least until we have tested this script to hell and back. Looks fine otherwise.

I removed the copying/backups  for now. Since all the files are just being moved, it will be more efficient if we
could just move them back to the original location (with exceptions like .fehbg which gets removed).

This can probably be implemented the same way in which  something like [trash-cli](https://github.com/andreafrancia/trash-cli) is able to restore the trashed files
to it's original location.

(force push 2: Added a fix to delete directories only if the contents were moved)

pystardust commented on tidytilde:
```diff
@@ -1,17 +1,88 @@
+	if [[ "$1" = "-a" ]]; then
+		shift
+		printf '\tAlias needs to be set. '
+		echo "alias $1='$2'" >> "$TIDYTILDE_COMMANDS"
```
use 
```bash
printf '%s\n' "alias $1='$2'" >>
```
If `$2` contained a `\n` then `echo` would print a new line there

pystardust replied:
similarly to all other echo's with variables

pystardust commented on tidytilde:
```diff
@@ -1,17 +1,88 @@
+	# exists, 'mv -i' will prompt the user for overwriting. If "$1" is a directory
+	# and the destination is an already existing dirctory, it will move the
+	# contents of the dotdir into the existing destination directory. If only "$1"
+	# is passed, it will remove it from $HOME.
```
if only `$1` being passed removing seems a little confusing. It may lead to some confusions. Maybe it would be better to have a remove_file function

pystardust commented on tidytilde:
```diff
@@ -1,17 +1,88 @@
+		if
+			[[ -d "$destination" ]]
+		then
+			[[ -n "$(ls -A "$dotfile")" ]] && mv -i "$dotfile"/* "$destination" || printf '\tRemoving empty directory\n'
```
since we are using bash, look at the `nullglob` shopt built in
```bash
shopt -s nullglob
files=( "$dotfile"/* )
if (( ${#files[@]} == 0 )); then
     mv ....
fi
```

pystardust replied:
null glob returns no objects if glob fails to match anyfile

Mudskipper875 committed:
Add a function for removing files and some fixes

Add a separate function for removing files. Use `$#` to test the number
of arguments in a function. Replace `echo` with `printf`. Use `nullglob`
to determine if directory is empty before removing it.

Mudskipper875 commented:
I've made the suggested changes.
